tests/test_spherePoint.py

```diff
@@ -762,48 +764,60 @@ def testRotatedAlias(self):
-
-        with self.assertRaises(pexEx.DomainError):
-            northPole.offset(-90.0*degrees, 10.0*degrees)
-        with self.assertRaises(pexEx.DomainError):
-            southPole.offset(90.0*degrees, 0.1*degrees)
-        with self.assertRaises(pexEx.InvalidParameterError):
-            nonSingularity.offset(90.0*degrees, -0.1*degrees)
-
     def testOffsetValue(self):
         """Test if offset() returns the expected value.
         """
         # This should cover arcs over the meridian, across the pole, etc.
         for lon1, lat1 in self._dataset:
             point1 = SpherePoint(lon1, lat1)
-            if point1.atPole():
-                continue
             for lon2, lat2 in self._dataset:
                 if lon1 == lon2 and lat1 == lat2:
                     continue
                 point2 = SpherePoint(lon2, lat2)
                 bearing = point1.bearingTo(point2)
                 distance = point1.separation(point2)
 
+                # offsetting point1 by bearing and distance should produce the same result as point2
                 newPoint = point1.offset(bearing, distance)
                 self.assertIsInstance(newPoint, SpherePoint)
+                self.assertSpherePointsAlmostEqual(point2, newPoint)
+                if newPoint.atPole():
+                    self.assertAnglesAlmostEqual(newPoint.getLongitude(), 0*degrees)
+
+                # measuring the separation and bearing from point1 to the new point
+                # should produce the requested separation and bearing
+                measuredDistance = point1.separation(newPoint)
+                self.assertAnglesAlmostEqual(measuredDistance, distance)
+                if abs(measuredDistance.asDegrees() - 180) > 1e-5:
+                    # The two points are not opposite each other on the sphere,
+                    # so the bearing has a well defined value
+                    measuredBearing = point1.bearingTo(newPoint)
+                    self.assertAnglesAlmostEqual(measuredBearing, bearing)
+
+                # offset by a negative amount in the opposite direction should produce the same result
+                newPoint2 = point1.offset(bearing + 180 * degrees, -distance)
+                self.assertIsInstance(newPoint2, SpherePoint)
+                self.assertSpherePointsAlmostEqual(newPoint, newPoint2)
+
                 if point1.isFinite() and point2.isFinite():
                     if not point2.atPole():
                         self.assertAnglesAlmostEqual(
                             point2.getLongitude(), newPoint.getLongitude())
+                        self.assertAnglesAlmostEqual(
+                            point2.getLongitude(), newPoint2.getLongitude())
                     self.assertAnglesAlmostEqual(
                         point2.getLatitude(), newPoint.getLatitude())
+                    self.assertAnglesAlmostEqual(
+                        point2.getLatitude(), newPoint2.getLatitude())
```

Comment:
These assertions are redundant with your use of `assertSpherePointsAlmostEqual`, above.

Reply:
I disagree. assertSpherePointsAlmostEqual tests angular separation, and thus cannot tell the difference between two SpherePoints at (or very near) the pole that have different longitudes.

Reply:
Then can you add a comment to that effect?